linkerd/app/outbound/src/switch_logical.rs

```diff
@@ -52,14 +53,28 @@ impl<S> Outbound<S> {
                                 tracing::debug!("Profile describes a logical service");
                                 return Ok(svc::Either::B(Logical::new(logical_addr, rx)));
                             }
+
+                            // If there was a profile but it didn't include an
+                            // endpoint or logical address, create a bare
+                            // endpoint from the original destination address,
+                            // using the profile-provided opaqeuness. This
+                            // mostly applies to external targets that may be
```

Comment:
the "mostly" here is a little bit unclear to me --- reading this, i would want to know whether there are any other cases.